oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/mongo/MongoDocumentStore.java

```diff
@@ -1470,6 +1487,18 @@ public <T extends Document> boolean create(Collection<T> collection, List<Update
                 }
                 insertSuccess = true;
                 return true;
+            } catch (BsonMaximumSizeExceededException e) {
+                T doct = null;
+                for (T doc : docs) {
+                    doct = doc;
+                    // doc.getMemory()/2 - converting from UTF-16 to UTF-8
+                    if (doc.getMemory()/2 > SIZE_LIMIT) {
+                        LOG.error("Failed to create the document with Id={} has size={}" +
+                                        " with BsonMaximumSizeExceededException message '{}",
+                                doc.getId(), doc.getMemory()/2, e.getMessage());
+                    }
+                }
+                throw handleException(e, collection, doct.getId());
```

Comment:
AFAICT, this method should return `false` here instead of throwing the exception.
@stefan-egli wdyt?

Reply:
throwing exception looks fine, why would you return `false`?

Reply:
+1, I was not looking at the whole context. return `false` is correct